chromeapp/app/scripts/dnssd/dns-controller.js

```diff
@@ -0,0 +1,585 @@
+/*jshint esnext:true*/
+/* globals Promise */
+'use strict';
+
+var chromeUdp = require('../chrome-apis/udp');
+var dnsUtil = require('./dns-util');
+var dnsPacket = require('./dns-packet');
+var byteArray = require('./byte-array');
+var dnsCodes = require('./dns-codes');
+var qSection = require('./question-section');
+
+/**
+ * This module maintains DNS state and serves as the DNS server. It is
+ * responsible for issuing DNS requests.
+ */
+
+var DNSSD_MULTICAST_GROUP = '224.0.0.251';
```

Comment:
Please add comments for these constants since if they are, e.g., from the RFC it is important that someone not tweak them unknowingly.